fastlane/Fastfile

```diff
@@ -0,0 +1,242 @@
+  build_path: Constants.BUILD_PATH,
+  firebase_token: ENV['FIREBASE_CLI_TOKEN']
+)
+
+keychain_name = 'github_action_keychain'
+keychain_password = 'password'
+
+before_all do
+  ensure_bundle_exec
+end
+
+default_platform(:ios)
+
+platform :ios do
+  desc 'Sync development match signing'
+  lane :sync_development_signing do
+    match(type: 'development', readonly: true)
+  end
+
+  desc 'Sync AdHoc match signing'
+  lane :sync_adhoc_signing do
+    match(type: 'adhoc', readonly: true)
+  end
+
+  desc 'Sync Adhoc match signing on CI'
+  lane :sync_adhoc_signing_ci do
+    create_ci_keychain
+    match(
+        type: 'adhoc',
+        keychain_name: keychain_name,
+        keychain_password: keychain_password,
+        app_identifier: [Constants.BUNDLE_ID_STAGING],
+        readonly: true
+    )
+  end
+
+  desc 'Sync Adhoc Production match signing on CI'
+  lane :sync_adhoc_production_signing_ci do
+    create_keychain(
+        name: keychain_name,
+        password: keychain_password,
+        default_keychain: true,
+        unlock: true,
+        timeout: 3600,
+        lock_when_sleeps: false
+    )
+
+    match(
+        type: 'adhoc',
+        keychain_name: keychain_name,
+        keychain_password: keychain_password,
+        app_identifier: [Constants.BUNDLE_ID_PRODUCTION],
+        readonly: true
+    )
+  end
+
+  desc 'Sync AppStore match signing on CI'
+  lane :sync_appstore_signing_ci do
+    create_ci_keychain
+    match(
+        type: 'appstore',
+        keychain_name: keychain_name,
+        keychain_password: keychain_password,
+        app_identifier: [Constants.BUNDLE_ID_PRODUCTION],
+        readonly: true
+    )
+  end
+
+  desc 'Register new devices'
+  lane :register_new_device do
+    device_name = prompt(text: 'Enter the device name: ')
+    device_udid = prompt(text: 'Enter the device UDID: ')
+    device_hash = {}
+    device_hash[device_name] = device_udid
+    register_devices(devices: device_hash)
+    match(force: true)
+  end
+
+  # Deploy
+
+  desc 'build develop build'
+  lane :build_develop do
+    builder.build_develop
+  end
+
+  # Firebase
+
+  desc 'Build and upload staging app to firebase'
+  lane :build_and_upload_staging_app do
+    ENV['PRODUCT_NAME'] = Constants.PRODUCT_NAME_STAGING
+    ENV['FIREBASE_APP_ID'] = Constants.FIREBASE_APP_ID_STAGING
+    ENV['GSP_NAME'] = Constants.GSP_STAGING
+
+    set_app_version
+    bump_build
+    
+    builder.build_app_ad_hoc(
+      Constants.SCHEME_NAME_STAGING,
+      Constants.PRODUCT_NAME_STAGING,
+      Constants.BUNDLE_ID_STAGING
+    )
+    upload_build_to_firebase
+    upload_build_dsym_to_firebase
+  end
+
+  desc 'Build and upload Production app to firebase'
+  lane :build_and_upload_production_app do
+    ENV['PRODUCT_NAME'] = Constants.PRODUCT_NAME_PRODUCTION
+    ENV['FIREBASE_APP_ID'] = Constants.FIREBASE_APP_ID_PRODUCTION
+    ENV['GSP_NAME'] = Constants.GSP_PRODUCTION
+    ENV['TESTER_GROUPS'] = Constants.FIREBASE_TESTER_GROUPS
```

Comment:
I'm not a fan of this (edit global variable), let's use parameters instead, for example:
```rb
product_name = Constants.PRODUCT_NAME_PRODUCTION
firebase_app_id = Constants.FIREBASE_APP_ID_PRODUCTION
upload_build_to_firebase(product_name: product_name, firebase_app_id: firebase_app_id)
```